src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -163,122 +163,165 @@ public void SeekSkipsEmptySegments()
 
-            Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment2, c1.GetObject());
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 0);
+
+            var start = buffer.Start;
+            Assert.True(buffer.TryGet(ref start, out var memory));
+            Assert.Equal(100, memory.Length);
+            Assert.True(buffer.TryGet(ref start, out memory));
+            Assert.Equal(0, memory.Length);
+            Assert.False(buffer.TryGet(ref start, out memory));
         }
 
         [Fact]
-        public void TryGetSkipsEmptyForNextUsingGetPositionWithOffset()
+        public void TryGetStopsAtEndWhenEndIsFistByteOfEmpty()
         {
             var bufferSegment1 = new BufferSegment<byte>(new byte[100]);
             BufferSegment<byte> bufferSegment2 = bufferSegment1.Append(new byte[0]);
-            BufferSegment<byte> bufferSegment3 = bufferSegment2.Append(new byte[0]);
-            BufferSegment<byte> bufferSegment4 = bufferSegment3.Append(new byte[100]);
 
-            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment4, 100);
-            SequencePosition c1 = buffer.GetPosition(0);
-            Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment1, c1.GetObject());
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 0);
 
-            ReadOnlyMemory<byte> data;
-            Assert.True(buffer.TryGet(ref c1, out data, advance: true));
+            var start = buffer.Start;
+            Assert.True(buffer.TryGet(ref start, out var memory));
+            Assert.Equal(100, memory.Length);
+            Assert.True(buffer.TryGet(ref start, out memory));
+            Assert.Equal(0, memory.Length);
+            Assert.False(buffer.TryGet(ref start, out memory));
+        }
 
-            Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment4, c1.GetObject());
+        [Fact]
+        public void EnumerableStopsAtEndWhenEndIsLastByteOfFull()
+        {
+            var bufferSegment1 = new BufferSegment<byte>(new byte[100]);
+            BufferSegment<byte> bufferSegment2 = bufferSegment1.Append(new byte[100]);
 
-            Assert.True(buffer.TryGet(ref c1, out data, advance: true));
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment1, 100);
 
-            Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(null, c1.GetObject());
+            List<int> sizes = new List<int>();
+            foreach (var memory in buffer)
+            {
+                sizes.Add(memory.Length);
+            }
 
-            Assert.False(buffer.TryGet(ref c1, out data, advance: true));
+            Assert.Equal(1, sizes.Count);
+            Assert.Equal(new [] { 100 }, sizes);
         }
 
         [Fact]
-        public void TryGetSkipsEmptyForNextUsingGetPositionWithOffsetAndOrigin()
+        public void EnumerableStopsAtEndWhenEndIsFistByteOfFull()
```

Comment:
Fist -> First, did you mean first segment?

Reply:
First Byte of Empty [Segment]